src/Adapter/Carrier/CommandHandler/AddCarrierHandler.php

```diff
@@ -63,10 +63,16 @@ public function handle(AddCarrierCommand $command): CarrierId
         $carrier->position = $command->getPosition();
         $carrier->active = $command->getActive();
         $carrier->delay = $command->getLocalizedDelay();
+        $carrier->max_width = $command->getMaxWidth();
+        $carrier->max_height = $command->getMaxHeight();
+        $carrier->max_weight = $command->getMaxWeight();
+        $carrier->max_depth = $command->getMaxDepth();
 
         $this->carrierValidator->validate($carrier);
 
         $carrierId = $this->carrierRepository->add($carrier);
+        $this->carrierValidator->validateGroupsExist($command->getGroupAccess());
+        $this->carrierRepository->setGroupsAccess($this->carrierRepository->get($carrierId), $command->getGroupAccess());
```

Comment:
Why do you fetch the object that is being created? It costs an extra query that is useless

Reply:
because you told me not to fetch it in the repo and to take a carrier as a param. But it has to be fetched at some point since as you see line 73 the add returns an Id and not a carrier object